### PR TITLE
Add example for external UI

### PR DESCRIPTION
**Pull request by TodePond**
This PR adds an example for how to add external UI components.

It's something that I've been asked for help with in the past, and it's something I wish more consumers would do when using tldraw inline - it would help the editor feel less cramped. Therefore, I'd like to have it is as an example we can point people to.

2024-02-19 at 11 53 18 - Maroon Boar

> Some other context:
> We've talked about giving this sorta thing more first-class support in the future, eg: with a `useEditorInstance` hook or something.

Closes TLD-2128

### Change Type

- [x] `documentation` — Changes to the documentation only[^2]

[^1]: publishes a `patch` release, for devDependencies use `internal`
[^2]: will not publish a new version

### Test Plan

1. Try out the External UI example.
2. Make sure you can change tool by pressing the toolbar buttons.
3. Make sure you can change tool by pressing keys (eg: d, e)

- [ ] Unit Tests
- [ ] End to end tests

### Release Notes

- Docs: Added external UI example.

### CODE REVIEW

File: apps/examples/src/examples/image-component/TldrawImageExample.tsx [hunk 1]
```diff
@@ -100,8 +100,8 @@ and viewing it.
 
 [2] You can specify which page to display by using the `pageId` prop. By
     default, the first page is shown.
-	
+
 [3] You can customize the appearance of the image by passing other props to the
-		`TldrawImage` component. For example, you can toggle the background, set the
-		dark mode, and specify the viewport bounds.
+        `TldrawImage` component. For example, you can toggle the background, set
+        the dark mode, and specify the viewport bounds.
```
Comment (TodePond) [inline apps/examples/src/examples/image-component/TldrawImageExample.tsx:106]: unrelated scout cleaning

File: apps/examples/src/examples/external-ui/ExternalUiExample.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Store the editor in some state when the `Tldraw` component mounts.
+
+[2] Pass the editor to your external UI component. In this example, we're using
+the same component as in the `custom-ui` example, but you could use any UI you
+want. Check out that example for more info on how , including how to make the UI
+reactive.
+
+This example passes the editor down as a prop, but you could also use a context.
+
+Check out the `custom-ui` example for more information on how to create custom
```
Comment (Taha-Hassan-Git) [inline apps/examples/src/examples/external-ui/ExternalUiExample.tsx:107]: Is this supposed to be in twice?

**Review by Taha-Hassan-Git — APPROVED** [review]
Looks great! Very simple and well-explained. Just one minor comment.
👍

File: apps/examples/src/examples/external-ui/ExternalUiExample.tsx (hunk 2 above)
Comment (TodePond) [reply apps/examples/src/examples/external-ui/ExternalUiExample.tsx]: oop good spot, removed the extra line!

**Pull request merged**